### PR TITLE
Add asserts to SIL Instruction Builder functions that should only take loadable types

### PR DESCRIPTION
**Pull request by shajrawi**
radar rdar://problem/33767770

### CODE REVIEW

**Comment by shajrawi** [comment]
@swift-ci Please smoke test and merge

**Comment by shajrawi** [comment]
@swift-ci Please clean smoke test Linux

File: include/swift/SIL/SILBuilder.h [hunk 1]
```diff
@@ -980,6 +980,8 @@ class SILBuilder {
   RetainValueInst *createRetainValue(SILLocation Loc, SILValue operand,
                                      Atomicity atomicity) {
     assert(isParsing || !getFunction().hasQualifiedOwnership());
+    assert(!SILModuleConventions(getModule()).useLoweredAddresses() ||
```
Comment (slavapestov) [inline include/swift/SIL/SILBuilder.h]: This disjunction appears in many places in SIL and SILGen. Can we factor it out into a new method on SILType?

Comment (shajrawi) [reply include/swift/SIL/SILBuilder.h]: @atrick what do you think?
This is part of Andy's work on semantic SIL. Makes sense to me that it is a module's calling convention / an address-only type can be used as a value before lowering with that enabled.

Comment (atrick) [reply include/swift/SIL/SILBuilder.h]: In the general case, we want a single idiom `SILModuleConventions(getModule()).useLoweredAddresses()` to search for and remove once that mode is no longer supported in whatever SIL pass is using it.

In this case, it's in the SILBuilder itself, and we probably still want to use the SILBuilder after address lowering. So, in this case Slava's suggestion makes sense... but only in the SILBuilder. Not sure what to call it. `type.isLoadableOrLowered()`?

Comment (shajrawi) [reply include/swift/SIL/SILBuilder.h]: Done. added `type.isLoadableOrLowered` and used it in the builder

**Comment by shajrawi** [comment]
@swift-ci Please smoke test and merge

**Comment by shajrawi** [comment]
@swift-ci Please smoke test and merge

**Pull request merged**